Automatically deploy clickable HTML to GitHub Pages

As mentioned in #14, this is quite easy to do. However, an access token needs to be generated.

You can do this by navigating to GitHub settings > Developer settings > Personal access tokens. Just the public repository access privilege should be sufficient.

Then, navigate to the repository settings and then to the "Secrets" section. There you can add this as a secret, and it will be stored in an encrypted way. Then you need to pass this to the deployment action when you are calling it. Here is [my workflow](https://github.com/ayberkt/formal-topology-in-UF/blob/5969da5cafc04ced018210b2cbe30444df4d7cfa/.github/workflows/main.yml#L1-L31) that does this.

Honestly, I think this is an unnecessarily cumbersome procedure. On GitLab, all of this takes literally no more than five-six lines of configuration code as their HTML hosting machinery is itself a kind of "action" and you don't need to give any access or anything like that. However, GitHub is the popular option so it's easier to just stick with it.

I tried turning debug info on and this is what is printed to the logs:
```

2020-05-16T11:34:46.0657910Z ##[group]Run JamesIves/github-pages-deploy-action@releases/v3
2020-05-16T11:34:46.0658364Z with:
2020-05-16T11:34:46.0658684Z   ACCESS_TOKEN: ***
2020-05-16T11:34:46.0658763Z   BRANCH: gh-pages
2020-05-16T11:34:46.0659014Z   FOLDER: html
2020-05-16T11:34:46.0659280Z ##[endgroup]
2020-05-16T11:34:46.1899409Z Checking configuration and starting deployment… 🚦
2020-05-16T11:34:46.1912285Z Deploying using Access Token… 🔑
2020-05-16T11:34:46.1914504Z Configuring git…
2020-05-16T11:34:46.1991775Z [command]/usr/bin/git init
2020-05-16T11:34:46.2084724Z Reinitialized existing Git repository in /home/runner/work/cat/cat/.git/
2020-05-16T11:34:46.2141438Z [command]/usr/bin/git config user.name fredefox
2020-05-16T11:34:46.2200441Z [command]/usr/bin/git config user.email fredefox@users.noreply.github.com
2020-05-16T11:34:46.2258537Z [command]/usr/bin/git remote rm origin
2020-05-16T11:34:46.2328556Z [command]/usr/bin/git remote add origin https://***@github.com/fredefox/cat.git
2020-05-16T11:34:46.2387925Z [command]/usr/bin/git fetch
2020-05-16T11:34:46.7365950Z From https://github.com/fredefox/cat
2020-05-16T11:34:46.7368544Z  * [new branch]      abstract-funext   -> origin/abstract-funext
2020-05-16T11:34:46.7368818Z  * [new branch]      fredefox/gh-pages -> origin/fredefox/gh-pages
2020-05-16T11:34:46.7369072Z  * [new branch]      fredefox/travis   -> origin/fredefox/travis
2020-05-16T11:34:46.7370460Z  * [new branch]      gh-pages          -> origin/gh-pages
2020-05-16T11:34:46.7370878Z  * [new branch]      master            -> origin/master
2020-05-16T11:34:46.7371101Z  * [new branch]      try-wrapped-paths -> origin/try-wrapped-paths
2020-05-16T11:34:46.7540964Z  * [new tag]         0.0.1             -> 0.0.1
2020-05-16T11:34:46.7541648Z  * [new tag]         1.0.0             -> 1.0.0
2020-05-16T11:34:46.7542171Z  * [new tag]         1.0.1             -> 1.0.1
2020-05-16T11:34:46.7542657Z  * [new tag]         1.1.0             -> 1.1.0
2020-05-16T11:34:46.7543138Z  * [new tag]         1.2.0             -> 1.2.0
2020-05-16T11:34:46.7543626Z  * [new tag]         1.3.0             -> 1.3.0
2020-05-16T11:34:46.7544107Z  * [new tag]         1.4.0             -> 1.4.0
2020-05-16T11:34:46.7544564Z  * [new tag]         1.4.1             -> 1.4.1
2020-05-16T11:34:46.7545050Z  * [new tag]         1.5.0             -> 1.5.0
2020-05-16T11:34:46.7546510Z  * [new tag]         1.6.0             -> 1.6.0
2020-05-16T11:34:46.8730768Z Fetching submodule libs/agda-stdlib
2020-05-16T11:34:47.0713083Z Fetching submodule libs/cubical
2020-05-16T11:34:47.4210696Z fatal: remote error: upload-pack: not our ref 0d3f02e68297e940227137beac45fc1bce6e2bea
2020-05-16T11:34:47.7142770Z fatal: remote error: upload-pack: not our ref 4493cf249a1648be2ad365fe94ece337bfbcb5d9
2020-05-16T11:34:47.7215624Z Errors during submodule fetch:
2020-05-16T11:34:47.7215893Z 	libs/cubical
2020-05-16T11:34:47.7216327Z 	libs/agda-stdlib
2020-05-16T11:34:47.7257861Z ##[error]There was an error initializing the repository: The process '/usr/bin/git' failed with exit code 1 ❌
2020-05-16T11:34:47.7267482Z failed
2020-05-16T11:34:47.7269681Z Deployment Failed ❌
2020-05-16T11:34:47.7272095Z ::set-env name=DEPLOYMENT_STATUS::failed
```
So it appears to have something to do with the submodules that I'm using. I suspect some of the github actions are messing with the git index in some way. It's kinda difficult to figure out though given that it's not happening on my machine...

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -16,9 +16,15 @@ jobs:
         main-file: src/Cat.agda
         source-dir: .
         unsafe: true
-    - name: Upload HTML
+    - name: Upload HTML as build artifact
       id: html-upload
       uses: actions/upload-artifact@v1
       with:
         name: html
         path: html
+    - name: Upload HTML to Github Pages
+      uses: fredefox/github-pages-deploy-action@9913f79fa1eda103201056c00c8e1897ac4f921e
+      with:
+        ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+        BRANCH: gh-pages
+        FOLDER: html

--- a/README.md
+++ b/README.md
@@ -1,16 +1,16 @@
 Description
 ===========
-This project aims to formalize some parts of category theory using cubical agda
-&mdash; an extension to agda permitting univalence.  To learn more about this
-[read the docs](https://agda.readthedocs.io/en/latest/language/cubical.html).
+This project aims to formalize some parts of category theory using
+Cubical Agda &mdash; an extension to agda permitting univalence.  To
+learn more about Cubical Agda [read the
+docs](https://agda.readthedocs.io/en/latest/language/cubical.html).
 
 This project draws a lot of inspiration from [the
 HoTT-book](https://homotopytypetheory.org/book/).
 
 If you want more information about this project, then you're in luck.
-This is my masters thesis.  Go ahead and read it
-[here](http://web.student.chalmers.se/~hanghj/papers/univalent-categories.pdf)
-or alternative like so:
+This is my masters thesis.  It can be accessed here
+[here](https://hdl.handle.net/20.500.12380/256404) or build like so:
 
     cd doc/
     make

